plugins/quota/rest.go

```diff
@@ -2,29 +2,125 @@ package quota
 
 import (
 	"net/http"
+	"os"
+	"path"
 
+	"github.com/gluster/glusterd2/glusterd2/daemon"
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
 	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	volgen "github.com/gluster/glusterd2/glusterd2/volgen2"
 	"github.com/gluster/glusterd2/glusterd2/volume"
 	"github.com/gluster/glusterd2/pkg/api"
 	"github.com/gluster/glusterd2/pkg/errors"
 	"github.com/gorilla/mux"
 )
 
-func commonValidator(w http.ResponseWriter, r *http.Request) {
-	ctx := r.Context()
-	volname := mux.Vars(r)["volname"]
-	v, err := volume.GetVolume(volname)
+//QuotadStart is the function to start the quota daemon
+func QuotadStart(c transaction.TxnCtx) error {
+	quotadDaemon, err := NewQuotad()
 	if err != nil {
-		restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolNotFound.Error(), api.ErrCodeDefault)
-	} else if v.State != volume.VolStarted {
-		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrVolNotStarted.Error(), api.ErrCodeDefault)
+		return err
+	}
+	// Create pidfile dir if not exists
+	err = os.MkdirAll(path.Dir(quotadDaemon.pidfilepath), os.ModeDir|os.ModePerm)
+	if err != nil {
+		return err
 	}
+	// Create logFiledir dir
+	err = os.MkdirAll(path.Dir(quotadDaemon.logfilepath), os.ModeDir|os.ModePerm)
+	if err != nil {
+		return err
+	}
+	err = daemon.Start(quotadDaemon, true)
+	return nil
+}
+
+func storeVolume(c transaction.TxnCtx) error {
```

Comment:
This can be reused from existing transaction step. See below comment

Reply:
done